src/NServiceBus.Core/Persistence/Learning/SagaPersister/LearningSagaIdGenerator.cs

```diff
@@ -0,0 +1,25 @@
+namespace NServiceBus
+{
+    using System;
+    using Sagas;
+
+    class LearningSagaIdGenerator : ISagaIdGenerator
+    {
+        public Guid Generate(SagaIdGeneratorContext context)
+        {
+            if (context.CorrelationProperty == SagaCorrelationProperty.None)
+            {
+                throw new NotSupportedException("The development saga persister doesn't support custom saga finders.");
```

Comment:
I had this code

```
if (metadata.Finders.Any(f => f.Properties.ContainsKey("custom-finder-clr-type")))
                {
                    throw new NotSupportedException("The development saga persister doesn't support custom saga finders.");
                }
```

Then I realized we can't do this. It is perfectly reasonable to have multiple sagas in the same assembly even with custom finders but only ever route messages to an endpoint that hit the sagas that don't have a custom finder. By coding this into the learning persister we would prevent the startup of such an endpoint and forcing to split the assemblies. Is that something we really want to do?